### PR TITLE
WIP - Validate Content data in the Stages API

### PR DESCRIPTION
**Pull request by fabricio-aguiar**
https://pulp.plan.io/issues/5927
closes #5927

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/en/3.0/nightly/contributing/pull-request-walkthrough.html

### CODE REVIEW

**Comment by daviddavis** [comment]
Your code looks good. My concern here though is that this code is magical in that Content is automatically matched to a serializer without any input from the plugin writer. It's possible that plugin may have more than one serializer for a Content model depending on different contexts (like in pulp_rpm we have two for Package).

**Comment by fabricio-aguiar** [comment]
> Your code looks good. My concern here though is that this code is magical in that Content is automatically matched to a serializer without any input from the plugin writer. It's possible that plugin may have more than one serializer for a Content model depending on different contexts (like in pulp_rpm we have two for Package).

I agree, do you think the solution would be: require plugin writer to specify the serializer on DeclarativeContent?

File: pulpcore/plugin/stages/content_stages.py [hunk 1]
```diff
@@ -1,13 +1,54 @@
...
+                data[f.name] = list(f.value_from_object(content).values_list('pk', flat=True))
+        else:
+            data[f.name] = f.value_from_object(content)
+
+    if issubclass(serializer, SingleArtifactContentSerializer):
+        data["relative_path"] = data.get("relative_path", ".")
+        data["file"] = data.get("file", files.PulpTemporaryUploadedFile("name", "", 3, ""))
+
+    # avoiding deferred_validate on SingleArtifactContentUploadSerializer
+    serializer = serializer(data=data, context={"request": {}})
```
Comment (fabricio-aguiar) [inline pulpcore/plugin/stages/content_stages.py]: I'm not happy with these tricks to validating SingleArtifactContentSerializer and SingleArtifactContentUploadSerializer

**Comment by bmbouter** [comment]
I'd like to help with this when I get back from PTO. I think this PR is too fragile to merge.

File: pulpcore/plugin/stages/content_stages.py (hunk 1 above)
Comment (fabricio-aguiar) [reply pulpcore/plugin/stages/content_stages.py]: Alternative:
https://github.com/pulp/pulpcore/compare/master...fabricio-aguiar:5927.2

**Comment by fabricio-aguiar** [comment]
Closing in favor of: https://github.com/pulp/pulpcore/pull/532

**Pull request closed without merging**